src/librustc/ty/layout.rs

```diff
@@ -2327,158 +2327,6 @@ where
     }
 }
 
-impl<'a> HashStable<StableHashingContext<'a>> for Variants {
-    fn hash_stable(&self, hcx: &mut StableHashingContext<'a>, hasher: &mut StableHasher) {
-        use crate::ty::layout::Variants::*;
-        mem::discriminant(self).hash_stable(hcx, hasher);
-
-        match *self {
-            Single { index } => {
-                index.hash_stable(hcx, hasher);
-            }
-            Multiple {
-                ref discr,
-                ref discr_kind,
-                discr_index,
-                ref variants,
-            } => {
-                discr.hash_stable(hcx, hasher);
-                discr_kind.hash_stable(hcx, hasher);
-                discr_index.hash_stable(hcx, hasher);
-                variants.hash_stable(hcx, hasher);
-            }
-        }
-    }
-}
-
-impl<'a> HashStable<StableHashingContext<'a>> for DiscriminantKind {
-    fn hash_stable(&self, hcx: &mut StableHashingContext<'a>, hasher: &mut StableHasher) {
-        use crate::ty::layout::DiscriminantKind::*;
-        mem::discriminant(self).hash_stable(hcx, hasher);
-
-        match *self {
-            Tag => {}
-            Niche {
-                dataful_variant,
-                ref niche_variants,
-                niche_start,
-            } => {
-                dataful_variant.hash_stable(hcx, hasher);
-                niche_variants.start().hash_stable(hcx, hasher);
-                niche_variants.end().hash_stable(hcx, hasher);
```

Comment:
@eddyb Does this only hash the start and end and not stuff in the middle?

Reply:
Ah, I see. It's a `RangeInclusive`. @cjgillot You could impl `HashStable` for `RangeInclusive` and move this to a derive macro too.

Reply:
`niche_variants` is a `RangeInclusive<u128>`, `start` and `end` are all there is to it.
A `HashStable` impl for `RangeInclusive<T>` could also work, I guess.